src/v/redpanda/application.cc

```diff
@@ -1498,27 +1498,30 @@ void application::wire_up_and_start(::stop_signal& app_signal, bool test_mode) {
 
 void application::start_runtime_services(cluster::cluster_discovery& cd) {
     ssx::background = feature_table.invoke_on_all(
-      [this](features::feature_table& ft) -> ss::future<> {
-          try {
-              co_await ft.await_feature(features::feature::rpc_v2_by_default);
-              if (ss::this_shard_id() == 0) {
-                  vlog(_log.info, "Activating RPC protocol v2");
-              }
-              _connection_cache.local().set_default_transport_version(
-                rpc::transport_version::v2);
-          } catch (ss::abort_requested_exception&) {
-              // Shutting down
-              co_return;
-          } catch (...) {
-              // Should never happen, abort is the only exception that
-              // await_feature can throw, other than perhaps bad_alloc.
-              vlog(
-                _log.error,
-                "Unexpected error awaiting RPCv2 feature: {} {}",
-                std::current_exception(),
-                ss::current_backtrace());
-              co_return;
-          }
+      [this](features::feature_table& ft) {
+          return ft.await_feature(features::feature::rpc_v2_by_default)
+            .then([this] {
+                if (ss::this_shard_id() == 0) {
+                    vlog(_log.info, "Activating RPC protocol v2");
+                }
+                _connection_cache.local().set_default_transport_version(
+                  rpc::transport_version::v2);
+            })
+            .handle_exception([this](const std::exception_ptr& e) {
```

Comment:
You could have done:
```c++
        .handle_exception_type([](const ss::abort_requested_exception&){
            // Shutting down
        }).handle_exception(
```
Rather than the rethrow dance.

Reply:
oh yeh for sure. the rethrow option, tho, allowed the delta to be trivial to examine. i think if anyone wants to come back through and update this code now that we aren't crunched for time then that'd be a welcome change.